package/MDAnalysis/core/groups.py

```diff
@@ -1253,6 +1262,160 @@ def wrap(self, compound="atoms", center="com", box=None):
+        An AtomGroup containing all atoms of all fragments in the group ``ag``
+        can be created with::
+
+          all_frag_atoms = sum(ag.fragments)
+
+
+        See Also
+        --------
+        :func:`~MDAnalysis.lib.mdamath.make_whole`,
+        :meth:`wrap`,
+        :meth:`pack_into_box`,
+        :func:`~MDanalysis.lib.distances.apply_PBC`
+
+
+        .. versionadded:: 0.20.0
+        """
+        atoms = self.atoms
+        # bail out early if no bonds in topology:
+        if not hasattr(atoms, 'bonds'):
+            raise NoDataError("{}.unwrap() not available; this requires Bonds"
+                              "".format(self.__class__.__name__))
+        unique_atoms = atoms.unique
+        if reference is not None:
+            ref = reference.lower()
+            if ref  == 'com':
+                if not hasattr(self, 'masses'):
+                    raise NoDataError("Cannot perform unwrap with "
+                                      "reference='com', this requires masses.")
+            elif ref != 'cog':
+                raise ValueError("Unrecognized reference '{}'. Please use one "
+                                 "of 'com', 'cog', or None.".format(reference))
+        comp = compound.lower()
+        if comp not in ('fragments', 'group', 'residues', 'segments',
+                        'molecules'):
+            raise ValueError("Unrecognized compound definition: {}\nPlease use"
+                             " one of 'group', 'residues', 'segments', "
+                             "'molecules', or 'fragments'.".format(compound))
+        # The 'group' needs no splitting:
+        if comp == 'group':
+            positions = mdamath.make_whole(unique_atoms, inplace=False)
+            # Apply reference shift if required:
+            if reference is not None and len(positions) > 0:
+                if ref == 'com':
+                    masses = unique_atoms.masses
+                    total_mass = masses.sum()
+                    if np.isclose(total_mass, 0.0):
+                        raise ValueError("Cannot perform unwrap with "
+                                         "reference='com' because the total "
+                                         "mass of the group is zero.")
+                    refpos = np.sum(positions * masses[:, None], axis=0)
+                    refpos /= total_mass
+                else:  # ref == 'cog'
+                    refpos = positions.mean(axis=0)
+                target = distances.apply_PBC(refpos, self.dimensions)
+                positions += target - refpos
+        # We need to split the group into compounds:
+        else:
+            if comp == 'fragments':
+                compound_indices = unique_atoms.fragindices
+            elif comp == 'residues':
+                compound_indices = unique_atoms.resindices
+            elif comp == 'segments':
+                compound_indices = unique_atoms.segindices
+            elif comp == 'molecules':
```

Comment:
The codecov report suggests there's no case where control flow gets here and `comp != 'molecules'`. Indeed, there's already a `ValueError` guard above against unknown `comp` values, so you could just use `else` here safely & avoid the extraneous check / partial statement coverage.

Reply:
Ok let's crank the diff coverage up to 100%

Reply:
The commit history suggests you've already attacked similar cases elsewhere, and it is just barely worth doing, so that's up to you I think.

Reply:
Done. Let's see what CodeCov thinks about it... :crossed_fingers: